src/main/java/com/exasol/adapter/document/documentfetcher/files/parquet/ParquetColumnToMappingDefinitionConverter.java

```diff
@@ -25,28 +27,39 @@ private static String buildColumnName(final Type column) {
     MappingDefinition convert(final Type column) {
         final LogicalTypeAnnotation logicalTypeAnnotation = column.getLogicalTypeAnnotation();
         if (logicalTypeAnnotation != null) {
-            return new LogicalTypeConverter().convert(logicalTypeAnnotation, buildColumnName(column));
+            return new LogicalTypeConverter().convert(logicalTypeAnnotation,
+                    columnNameConverter.convertColumnName(column.getName()));
         } else {
-            final NonLogicalTypeConvertVisitor visitor = new NonLogicalTypeConvertVisitor();
+            final NonLogicalTypeConvertVisitor visitor = new NonLogicalTypeConvertVisitor(this.columnNameConverter);
             column.accept(visitor);
             return visitor.getResult();
         }
     }
 
     private static class NonLogicalTypeConvertVisitor implements TypeVisitor {
+        private final ColumnNameConverter columnNameConverter;
         private MappingDefinition result;
 
+        private NonLogicalTypeConvertVisitor(final ColumnNameConverter columnNameConverter) {
+            this.columnNameConverter = columnNameConverter;
+        }
+
         @Override
         public void visit(final GroupType groupType) {
             groupType.getLogicalTypeAnnotation();
             final Fields.FieldsBuilder fieldsBuilder = Fields.builder();
             for (final Type column : groupType.getFields()) {
-                final MappingDefinition columnMapping = new ParquetColumnToMappingDefinitionConverter().convert(column);
+                final MappingDefinition columnMapping = new ParquetColumnToMappingDefinitionConverter(
+                        columnNameConverter).convert(column);
                 fieldsBuilder.mapField(column.getName(), columnMapping);
             }
             this.result = fieldsBuilder.build();
         }
 
+        private String buildColumnName(final Type column) {
+            return columnNameConverter.convertColumnName(column.getName());
+        }
```

Comment:
Is this needed? Can't find any reference.....

Reply:
Yes. There was a static method with the same name. The method now needs the `columnNameConverter`, so I moved the method here.